### PR TITLE
fix(scylla-cql): reject lz4 frame bodies shorter than the 4-byte size prefix

### LINKED ISSUE #1639: LZ4 decompression panics on truncated input
The LZ4 branch in [`decompress`](https://github.com/scylladb/scylla-rust-driver/blob/9226fe6c0fbcdc0965cc274c5552ec47da465d33/scylla-cql/src/frame/mod.rs#L357-L366) calls `get_u32()` unconditionally to read the size prefix. A short compressed frame panics before the code can return `FrameBodyExtensionsParseError`.

```rust
#[test]
fn test_lz4_decompress_rejects_short_input() {
    assert!(decompress(&[], Compression::Lz4).is_err());
}
```

### PR DESCRIPTION
**Pull request by SAY-5**
Calling `Buf::get_u32()` on a short slice panicked inside bytes rather than producing `FrameBodyExtensionsParseError::Lz4DecompressError`. Validate the length up front so truncated server frames return a normal decode error.

Fixes: #1639

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved error handling and validation for LZ4-compressed data processing with clearer error messages for malformed or incomplete inputs

* **Tests**
  * Added unit tests for LZ4 decompression to validate proper handling of edge cases with insufficient data

* **Chores**
  * Updated core library dependencies to latest stable versions for enhanced stability and compatibility

[Review Change Stack](https://app.coderabbit.ai/change-stack/scylladb/scylla-rust-driver/pull/1642?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: scylla-cql/src/frame/mod.rs [hunk 1]
```diff
@@ -360,6 +360,14 @@ pub fn decompress(
...
     match compression {
         Compression::Lz4 => {
+            if comp_body.len() < std::mem::size_of::<u32>() {
+                return Err(FrameBodyExtensionsParseError::Lz4DecompressError(Arc::new(
+                    std::io::Error::new(
+                        std::io::ErrorKind::UnexpectedEof,
+                        "lz4 frame body is shorter than its 4-byte size prefix",
+                    ),
+                )));
+            }
```
Comment (Lorak-mmk) [inline scylla-cql/src/frame/mod.rs]: @wprzytula Requesting your review as well. I thought that LowLevelDeserializationError would be a better fit for the inner error, but see no good variant for that.
Lz4DecompressError stores dynamic error anyway, so there is no point in spending too much time on that (we can change the error later in 1.x). But let's decide what to do for now:
- Current version (io::Error)
- LLDE::TooFewBytesReceived, which has a message that is imo not good here
- New variant of LLDE (ehhh)
- LLDE::IoError - maybe this makes sense?

File: scylla-cql/src/frame/mod.rs [hunk 2]
```diff
@@ -360,6 +360,14 @@ pub fn decompress(
...
             let uncomp_len = comp_body.get_u32() as usize;
```
Comment (wprzytula) [inline scylla-cql/src/frame/mod.rs]: 🔧 Use `try_get_u32()` instead. This explicitly checks instead of asserting in `get_u32()`.

File: scylla-cql/src/frame/mod.rs (hunk 1 above)
Comment (wprzytula) [reply scylla-cql/src/frame/mod.rs]: `LLDE::IoError` makes sense.

**Review by wprzytula — CHANGES_REQUESTED** [review]

**Review by wprzytula — APPROVED** [review]

File: scylla-cql/Cargo.toml [hunk 3]
```diff
@@ -69,7 +69,7 @@ scylla-macros = { version = "=1.6.0", path = "../scylla-macros" }
 # AsyncRead trait used in read_response_frame
 tokio = { version = "1.40", features = ["io-util"] }
 # FrameSlice and other parts of public API
-bytes = "1.0.1"
+bytes = "1.10"
 # Tracing ids, CqlTimeuuid, ser/deser of CQL UUID
```
[the remark this answers is not in the document]
Comment (wprzytula) [reply scylla-cql/Cargo.toml:73]: We don't want `get_u32()` for sure.
One thing Copilot correctly noted, though: let's coordinate `bytes` dep version in all crates in the workspace.

File: scylla-cql/src/frame/mod.rs [hunk 4]
```diff
@@ -360,7 +361,16 @@ pub fn decompress(
...
         Compression::Lz4 => {
-            let uncomp_len = comp_body.get_u32() as usize;
+            let uncomp_len = comp_body
+                .try_get_u32()
+                .map_err(|_| {
+                    FrameBodyExtensionsParseError::Lz4DecompressError(Arc::new(
+                        LowLevelDeserializationError::IoError(Arc::new(std::io::Error::new(
+                            std::io::ErrorKind::UnexpectedEof,
+                            "lz4 frame body is shorter than its 4-byte size prefix",
+                        ))),
```
[the remark this answers is not in the document]
Comment (wprzytula) [reply scylla-cql/src/frame/mod.rs]: @Lorak-mmk Why didn't this error variant suit you?

[the remark this answers is not in the document]
Comment (Lorak-mmk) [reply scylla-cql/src/frame/mod.rs]: I didn't really like the error message: `#[error("Not enough bytes! expected: {expected}, received: {received}")]`. We are certainly expecting more bytes for lz4 frame, not just 4.

[the remark this answers is not in the document]
Comment (wprzytula) [reply scylla-cql/src/frame/mod.rs]: OK.

**Review by Lorak-mmk — APPROVED** [review]
